### PR TITLE
Feature/rspec - Code Review

### PR DESCRIPTION
**Pull request by rammazzoti2000**
This branch is for code review only.

In this branch has been: 
- Created test cases using `Rspec` framework
- Created variables for each test case using `let`
- Tested positive and negative scenarios
- Used rubocop to remove offenses

### CODE REVIEW

**Comment by euqueme** [comment]
Hello @rammazzoti2000 

I can't start the review because I only see commits performed by you but not from your coding partner @jaspreet-singh-sahota and as you know this is a collaborative project so please create another branch and make sure both students perform proportional commits before your next submission. 

If you worked on your own make sure to mark the checkbox in the review request. before submitting it again.

Happy coding! :rocket: 

Any question ask me on slack **@ Maru**

File: spec/enumerable_spec.rb [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+    it 'should throw an error if given an argument' do
+      expect { array.my_each_with_index('argument') }.to raise_error(ArgumentError)
+    end
+  end
+
+  describe '#my_select' do
+    arr_even = [1, 2, 3, 8]
+    arr_odd = [6, 11, 13]
+    odd = [11, 13]
+    array = [3, 5, 'A', 'B']
```
Comment (euqueme) [inline spec/enumerable_spec.rb]: [OPTIONAL]

- [x] you can also define the variables within your describe blocks using let

**Review by euqueme — APPROVED** [review]
> ### Pull Request under the review by Maria Eugenia Queme P. (TSE)

### Hello @rammazzoti2000 :handshake: 

### You've done a marvelous job, your project exceeds professionalism, claps to you for using other ways different from "to eq" or "not_to eq" to validate your positive and negative test cases this is a yes for me :clap: :clap: :clap: :heavy_check_mark: 

[OPTIONAL]

- [x] I won't stop you from moving forward for this but in the test cases you have for example

```ruby
expect (value).to eq(false)
```
you could also validate like this:
```ruby
expect(value).not_to eq(true)
```

### Keep up the good work :clinking_glasses: 

PROJECT STATUS: APPROVED :green_apple:

File: spec/enumerable_spec.rb (hunk 1 above)
Comment (rammazzoti2000) [reply spec/enumerable_spec.rb]: Thank you for advice, just made the suggested changes.

**Comment by rammazzoti2000** [comment]
made all optional required changes, thank you very much for your feedback.

**Pull request merged**